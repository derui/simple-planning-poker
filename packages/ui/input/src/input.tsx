import React from "react";
import { input } from "./style.css.js";

const _Input = function Input(props: React.HTMLProps<HTMLInputElement>, ref: React.Ref<HTMLInputElement> | undefined) {
  const { className: _, ...rest } = props;
  return <input ref={ref} {...rest} className={input} />;
};

export const Input: React.ForwardRefExoticComponent<
  Omit<React.HTMLProps<HTMLInputElement>, "ref"> & React.RefAttributes<HTMLInputElement>
> = React.forwardRef(_Input);
