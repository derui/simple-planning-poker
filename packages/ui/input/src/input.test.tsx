import { cleanup, render } from "@testing-library/react";
import { afterEach, expect, test } from "vitest";
import { Input } from "./input.js";

afterEach(cleanup);

test("should be able to render", () => {
  const ret = render(<Input />);

  expect(ret.container).toMatchSnapshot();
});
