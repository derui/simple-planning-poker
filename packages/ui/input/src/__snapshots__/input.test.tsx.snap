// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be able to render 1`] = `
<div>
  <input
    class="style_input__tb17r50"
  />
</div>
`;
